twopaneview/src/TwoPaneView.tsx

```diff
@@ -62,85 +59,63 @@ export class TwoPaneView extends Component<Props, State> {
   renderChildPanes() {
     const children = React.Children.toArray(this.props.children);
 
-    if (this.state.spanning) {
-      if (this.state.paneMode === PaneMode.Single || 
-        this.state.dims.height > this.state.dims.width  && this.state.panePriorityVerticalSpanning) {
-        if (this.state.panePriorityVerticalSpanning === PanePriority.Pane1) {
-          return this.renderPane1(this.getEntireSize());
-        }
-        else {
-          return this.renderPane2(this.getEntireSize());
-        }
+    const paneMode = this.getPaneMode();
+    if (paneMode === PaneMode.Auto) {
+      // TODO:  add logic for auto-detecting width > threshold
+      if (this.state.spanning) {
+        return this.renderBothPanes();
       }
-      return this.renderBothPanes();
+      return this.renderPaneWithPriority();
     }
-    if (this.state.paneMode === PaneMode.Dual) {
-      return this.renderDualPanes();
+    if (paneMode === PaneMode.Single) {
+      return this.renderPaneWithPriority();
     }
-    if (this.state.panePriority === PanePriority.Pane1) {
-      return this.renderPane1(this.getEntireSize());
+    if (paneMode === PaneMode.Double) {
+      return this.renderBothPanes();
     }
-    return this.renderPane2(this.getEntireSize());
   }
 
-  renderDualPanes() {
-    const children = React.Children.toArray(this.props.children);
-
-    const items = [];
-    if (children.length > 0) {
-      items.push(this.renderPane1(this.getLeftSize()));
+  renderPaneWithPriority() {
+    if (this.getPanePriority() === PanePriority.Pane1) {
+      return this.renderPane1();
     }
-
-
-    if (children.length > 1) {
-      items.push(this.renderPane2(this.getRightSize()));
+    else {
+      return this.renderPane2();
     }
-
-    return items;
   }
-
+  
   renderBothPanes() {
-    let horizontal = this.state.dims.width >= this.state.dims.height;
     const children = React.Children.toArray(this.props.children);
 
     const items = [];
-    if (children.length > 0) {
-      items.push(this.renderPane1(horizontal ? this.getLeftSize() : this.getTopSize()));
-    }
-
+    items.push(this.renderPane1());
     items.push(this.renderSeparator());
-
-    if (children.length > 1) {
-      items.push(this.renderPane2(horizontal ? this.getRightSize() : this.getBottomSize()));
-    }
+    items.push(this.renderPane2());
 
     return items;
```

Comment:
easier just to return array with all items

Reply:
I now have a conditional in the middle, so doing the ugly push seems necessary.

I admit I am not very experienced with javascript, open to further suggestions!


---
In reply to: [426023749](https://github.com/microsoft/react-native-dualscreen/pull/34#discussion_r426023749) [](ancestors = 426023749)